src/main/scala/io/iohk/ethereum/db/dataSource/EphemDataSource.scala

```diff
@@ -9,38 +9,48 @@ class EphemDataSource(var storage: Map[ByteBuffer, Array[Byte]]) extends DataSou
     * key.drop to remove namespace prefix from the key
     * @return key values paris from this storage
     */
-  def getAll(namespace: Namespace): Seq[(IndexedSeq[Byte], IndexedSeq[Byte])] =
-    storage.toSeq.map{case (key, value) => (key.array().drop(namespace.length).toIndexedSeq, value.toIndexedSeq)}
-
-  override def get(namespace: Namespace, key: Key): Option[Value] = storage.get(ByteBuffer.wrap((namespace ++ key).toArray)).map(_.toIndexedSeq)
-
-  override def update(namespace: Namespace, toRemove: Seq[Key], toUpsert: Seq[(Key, Value)]): DataSource = {
-    val afterRemoval = toRemove.foldLeft(storage)((storage, key) => storage - ByteBuffer.wrap((namespace ++ key).toArray))
-    val afterUpdate = toUpsert.foldLeft(afterRemoval)((storage, toUpdate) =>
-      storage + (ByteBuffer.wrap((namespace ++ toUpdate._1).toArray) -> toUpdate._2.toArray))
-    storage = afterUpdate
-    this
+  def getAll(namespace: Namespace): Seq[(IndexedSeq[Byte], IndexedSeq[Byte])] = synchronized {
+    storage.toSeq.map { case (key, value) => (key.array().drop(namespace.length).toIndexedSeq, value.toIndexedSeq) }
   }
 
-  override def clear: DataSource = {
-    storage = Map()
-    this
+  override def get(namespace: Namespace, key: Key): Option[Value] = {
+    storage.get(ByteBuffer.wrap((namespace ++ key).toArray)).map(_.toIndexedSeq)
   }
 
-  override def close(): Unit = ()
+  override def getOptimized(key: Array[Byte]): Option[Array[Byte]] = storage.get(ByteBuffer.wrap(key))
 
-  override def destroy(): Unit = ()
+  override def update(dataSourceUpdates: Seq[DataUpdate]): Unit = synchronized {
+    dataSourceUpdates.foreach {
+      case DataSourceUpdate(namespace, toRemove, toUpsert) =>
+        update(namespace, toRemove, toUpsert)
+      case DataSourceUpdateOptimized(toRemove, toUpsert) =>
+        updateOptimized(toRemove, toUpsert)
+    }
+  }
 
-  override def updateOptimized(toRemove: Seq[Array[Byte]], toUpsert: Seq[(Array[Byte], Array[Byte])]): DataSource = {
+  def update(namespace: Namespace, toRemove: Seq[Key], toUpsert: Seq[(Key, Value)]): Unit = synchronized {
+    val afterRemoval =
+      toRemove.foldLeft(storage)((storage, key) => storage - ByteBuffer.wrap((namespace ++ key).toArray))
+    val afterUpdate = toUpsert.foldLeft(afterRemoval)(
+      (storage, toUpdate) => storage + (ByteBuffer.wrap((namespace ++ toUpdate._1).toArray) -> toUpdate._2.toArray)
+    )
+    storage = afterUpdate
+  }
+
+  def updateOptimized(toRemove: Seq[Array[Byte]], toUpsert: Seq[(Array[Byte], Array[Byte])]): Unit = synchronized {
```

Comment:
Minor: should this and the update method (`def update(namespace: Namespace, toRemove: Seq[Key], toUpsert: Seq[(Key, Value)]): Unit`) be private?